src/invert/laplace/impls/cyclic/cyclic_laplace.cxx

```diff
@@ -240,12 +240,41 @@ FieldPerp LaplaceCyclic::solve(const FieldPerp& rhs, const FieldPerp& x0) {
                      &C1coef, &C2coef, &Dcoef,
                      false); // Don't include guard cells in arrays
       }
+      if (localmesh->periodicX and localmesh->firstX()) {
+        // Special case for kz = 0
+        // Need to exclude the kx=0,kz=0 mode
+        // - Set an arbitrary location to zero
+        // - Solve periodic tridiagonal system
+        // - Subtract average of result
+
+        a(0, 0) = 0.0;
+        b(0, 0) = 1.0;
+        c(0, 0) = 0.0;
+        bcmplx(0, 0) = 0.0;
+      }
```

Comment:
Should this go in `Laplace::tridagMatrix()`? That might mean other solvers (e.g. `pcr`) get a more sensible result if they're run with `periodicX`, although the code to subtract the x-average would have to go in each individual solver to fully support it.

Reply:
Thanks @johnomotani yes this is a good idea, now done.